### PR TITLE
Remove specialised historian registry type

### PR DESCRIPTION
**Pull request by sstone1**
## Checklist
 - [ ]  A link to the issue/user story that the pull request relates to
 - [ ]  How to recreate the problem without the fix
 - [ ]  Design of the fix
 - [ ]  How to prove that the fix works
 - [ ]  Automated tests that prove the fix keeps on working
 - [ ]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Issue/User story

As a user of a business network, I can view a record of all significant business events #55

## Steps to Reproduce

1.
2.
3.
4.

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [ ] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

## Design of the fix

We have a major version bump coming this week to introduce identity cards.
This is a good opportunity to clean up some of the debt in the historian (see #55 for details).

- Remove the `Historian` registry type, and make the historian work as per the identity registry. This means that an asset registry named `org.hyperledger.composer.system.HistorianRecord` will be used. This makes the runtime code simpler and the client code simpler. The registry type and ID are not externals, and are only exposed in the query language - it doesn't matter that they are assets - and this improves historian queries as the FROM clause can be dropped.
- Remove system types SystemRegistry, HistorianRegistry, and ResourceId, neither of which are used anywhere.

### CODE REVIEW

File: packages/composer-client/changelog.txt [hunk 1]
```diff
@@ -12,6 +12,9 @@
 # Note that the latest public API is documented using JSDocs and is available in api.txt.
 #
 
+Version 0.11.4 {24d2234e4b8ccafd57c303c90792ec31} 2017-08-18
```
Comment (davidkel) [inline packages/composer-client/changelog.txt]: should this be changed after a version bump has been done ?

File: packages/composer-client/lib/businessnetworkconnection.js [hunk 2]
```diff
@@ -182,7 +182,7 @@ class BusinessNetworkConnection extends EventEmitter {
 
     /**
      * Get a list of all existing participant registries.
-     * @exampleserializedResources
+     * @example
      * // Get all participant registries
      * var businessNetwork = new BusinessNetworkConnection();
      * return businessNetwork.connect('testprofile', 'businessNetworkIdentifier', 'WebAppAdmin', 'DJY27pEnl16d')
```
Comment (davidkel) [inline packages/composer-client/lib/businessnetworkconnection.js]: Should we have an issue to change these examples ?

File: packages/composer-common/lib/system.cto [hunk 3]
```diff
@@ -108,23 +102,15 @@ asset HistorianRecord identified by transactionId {
   --> Identity    identityUsed         optional
```
Comment (davidkel) [inline packages/composer-common/lib/system.cto]: should HistorianRecord extend asset in the same way Identity does ?

Comment (sstone1) [inline packages/composer-common/lib/system.cto]: The other way round - I fixed the problem which meant you had to explicitly extend Asset, so I'll remove the `extend Asset` from the other types.

File: packages/composer-client/lib/businessnetworkconnection.js (hunk 2 above)
Comment (sstone1) [inline packages/composer-client/lib/businessnetworkconnection.js]: Yep, have raised #1932.

File: packages/composer-client/changelog.txt (hunk 1 above)
Comment (sstone1) [inline packages/composer-client/changelog.txt]: Yeah, I'll take the hit for the version bump.

**Pull request merged**